channeldb/invoices.go

```diff
@@ -96,14 +91,40 @@ const (
 
 	// A set of tlv type definitions used to serialize invoice htlcs to the
 	// database.
+	//
+	// NOTE: A migration should be added whenever this list changes. This
+	// prevents against the database being rolled back to an older
+	// format where the surrounding logic might assume a different set of
+	// fields are known.
 	chanIDType       tlv.Type = 1
 	htlcIDType       tlv.Type = 3
 	amtType          tlv.Type = 5
 	acceptHeightType tlv.Type = 7
 	acceptTimeType   tlv.Type = 9
 	resolveTimeType  tlv.Type = 11
 	expiryHeightType tlv.Type = 13
-	stateType        tlv.Type = 15
+	htlcStateType    tlv.Type = 15
+
+	// A set of tlv type definitions used to serialize invoice bodiees.
+	//
+	// NOTE: A migration should be added whenever this list changes. This
+	// prevents against the database being rolled back to an older
+	// format where the surrounding logic might assume a different set of
+	// fields are known.
```

Comment:
:+1:
It is a no-op migration, hopefully that is clear.

Reply:
it is only a noop migration if the field is truly optional, in most cases that won't be the case i presume

Reply:
hmm, so in order to minimize migrations, what would we do with for example a new required int field? write a migration or define non-present as 0? i was hoping we could have fewer migrations now, but maybe not

Reply:
if it is required, idt there's a way around it. some things may be less critical and it would be okay to do an in-flight migration. in this case tho, these parameters affect invoice settling and i think qualify as required. i think it will have to be a case by case basis